app/controllers/user_profiles_controller.rb

```diff
@@ -3,29 +3,24 @@
 # Controller for create and edit UserProfile
 class UserProfilesController < ApplicationController
   before_action :set_info, only: %i[new create edit]
-  # before_action ... , only: %i[index new create edit update] # logged in?
+  before_action :signed_in, only: %i[index new create show edit update]
   # before_action ... , only: %i[edit update] # correct user?
 
   def index
     @user_profiles = UserProfile.all.page(params[:page])
   end
 
   def new
-    @user = User.new
-    # @user_profile = current_user.build_user_profile
-    @user_profile = @user.build_user_profile
+    if current_user.profile_registered?
+      redirect_to user_profiles_path # temp
```

Comment:
こちら一覧にリダイレクトしているように見えますが、編集にリダイレクトしてあげたほうがいいかもと思いました。

```suggestion
      redirect_to user_profile_path(current_user.user_profile)
```

かな？

Reply:
今回は更新編集プロフィールのコードを触ったことがないので、別issurでこうなる予定です。ありがとうございます。